content/blog/2022/06/Optimizing-Camel-K-Integration-Build-Time/index.md

```diff
@@ -0,0 +1,100 @@
+
+The effect of faster build times may be seen in the benchmark comparisons below. 
+
+### System Configuration 
+
+
+| Device         | Chip              | Total Number of Cores | Memory |
+| -------        | -----             | --------              |--------|
+|MacBook Air (M1)| Apple Sillicon M1 |   8 core CPU (4 performance and 4 efficiency), 7 Core GPU                |                  8GB
+
+### Environment Configuration 
+
+The tests are graded using the following configurations: 
+
+1. **Minikube Version:**  v1.25.2
+2. **Minimum no. of cores:** 2
+3. **Maximum no. of cores:** 4
+4. **Maximum memory used:** 3933 MB
+5. **Registry Addon:** Enabled 
+
+*PS: These metrics are exclusive to my system; they may differ for others.*
+
+### Evaluating Performing 
+
+1. Start a new minikube cluster with minimum permissible cores and maximum memory (as per docker desktop)
+    ```
+    minikube start --addons registry --cpus 2 --memory=3GB
+    ```
+2. Setup the Camel-K environment by installing the operator (Kamel)
+    ```
+    kamel install
+    ```
+3. Run a sample Integration in the dev mode
+    ```
+    kamel run --env MY_ENV_VAR="hello world" Env.java --dev
+    ```
+4. Get the build duration 
+    The build duration is ~6mins using default usage of cores available. 
+
+    ![default CPU usage build time](build-time-default.png)
+
+5. Delete the minikube cluster to reinstate the environment and avoid any usage of previously build Integration images. 
+    ```
+    minikube delete
+    ```
+6. Start a new minikube cluster with maximum permissible cores and maximum memory (as per docker desktop)
+    ```
+    minikube start --addons registry --cpus 4 --memory=3GB
+    ```
+7. Repeat steps 2 and 3. 
+8. Get the build duration
+    The build duration is ~2mins utilizing the maximum CPU cores available for my system. 
+
+    ![build time based on max CPU quota](optimized-build-time.png)
+
+#### Benchmark Comparison
+
+| Integration | Integration Kit Name | Build Time |CPUs |
+| -------- | -------- | -------- |----- |
+| Env.java     |  kit-caopeg968cvc73fie4t0   |   6m26.044974722s   | 2 | 
+|Env.java | kit-caort9bmr3lc73emiifo | 2m24.9183443385s | 4
+
+It is clear that the build time has decreased significantly; with more powerful systems (e.g., 16, 32 cores), the build time will be reduced to a few tens of seconds.
+
+### Leveraging the benefit 
+
+To utilise this newly available functionality and apply CPU constraints on operator deployment, select from the following example measures or consult the [Resource Management](https://camel.apache.org/camel-k/1.9.x/installation/advanced/resources.html#scheduling-infra-pod-resources) guide to modify it about their own needs.
```

Comment:
The link should be `/camel-k/1.9.x/installation/advanced/resources.html#scheduling-infra-pod-resources`.

Reply:
apologies, didn't know about the in-repo links. I've updated.